newrelic/resource_newrelic_nrql_drop_rule.go

```diff
@@ -149,11 +162,28 @@ func resourceNewRelicNRQLDropRuleDelete(ctx context.Context, d *schema.ResourceD
 
 	deleteInput := []string{ruleID}
 
-	_, err = client.Nrqldroprules.NRQLDropRulesDeleteWithContext(ctx, accountID, deleteInput)
+	deleted, err := client.Nrqldroprules.NRQLDropRulesDeleteWithContext(ctx, accountID, deleteInput)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
+	var diags diag.Diagnostics
+	if len(deleted.Failures) != 0 {
+		for _, resp := range deleted.Failures {
+			switch resp.Error.Reason {
+			case "GENERAL":
```

Comment:
Can we create a list of enums for these constants?

e.g. 

```go
type DropRuleErrorType string 

var DropRuleErrorTypes = struct {
	GENERAL DropRuleErrorType
	RULE_NOT_FOUND DropRuleErrorType
}{
	GENERAL: "GENERAL",
	RULE_NOT_FOUND: "RULE_NOT_FOUND",
}
```

Reply:
Do you think `INVALID_QUERY` should also be a part of this list too?

Reply:
It has [these](https://github.com/newrelic/newrelic-client-go/blob/main/pkg/nrqldroprules/types.go#L28) types returned out of which GENERAL is the only error that doesn't have a proper description.

Reply:
Oh nice, then let's use the types here instead of a raw string?

Something like this?
```go
switch resp.Error.Reason {
case nrqldroprules.NRQLDropRulesErrorReasonTypes.GENERAL:
	// ...
```

Reply:
ok